### PR TITLE
Remove AMP headers. Remove restrict use. Refactor tests to just use HC.

### PR DESCRIPTION
**Pull request by AlexVlx**
This is a rather mechanical first step, which removes the headers and the usage of C++AMP specific constructs whilst shifting everything to HC. This does outline a few things we've not implemented for HC such as tiled_extent::truncate, for example. These will be addressed with HC Next.

### CODE REVIEW

File: tests/Unit/Atomic/atomic_xor_global.cpp [hunk 1]
```diff
@@ -1,9 +1,9 @@
 // RUN: %cxxamp %s -o %t.out && %t.out
```
Comment (aaronenyeshi) [inline tests/Unit/Atomic/atomic_xor_global.cpp:1]: do we need to clean up the RUN line's cxxamp or leave it as it is?

File: tests/Unit/Design/pass_by_ref.cpp [hunk 2]
```diff
@@ -11,47 +11,47 @@
...
   }
  private:
-  Concurrency::array_view<int> a_, b_;
-  Concurrency::array_view<int>& c_;
+  hc::array_view<int> a_, b_;
+  hc::array_view<int>& c_;
 };
-void bar(void) restrict(amp,cpu) {
+void bar(void) [[cpu, hc]] {
   int* foo = reinterpret_cast<int*>(&myVecAdd::__cxxamp_trampoline);
```
Comment (aaronenyeshi) [inline tests/Unit/Design/pass_by_ref.cpp:27]: what is cxxamp_trampoline?

**Review by aaronenyeshi — APPROVED** [review]
LGTM - Thanks Alex. I haven't read everything in detail, this patch is very long. Should we remove: https://github.com/RadeonOpenCompute/hcc/wiki#c-amp

File: tests/Unit/Design/pass_by_ref.cpp (hunk 2 above)
Comment (AlexVlx) [reply tests/Unit/Design/pass_by_ref.cpp:27]: It's a compiler generated entry point for the kernel (long-ish story). It's going out as well as part of the AMP removal / HC Next refactoring, but that's part of another PR.

File: tests/Unit/Atomic/atomic_xor_global.cpp (hunk 1 above)
Comment (AlexVlx) [reply tests/Unit/Atomic/atomic_xor_global.cpp:1]: Eventually yes, we'll remove that one.

File: tests/Unit/Design/pass_by_ref.cpp (hunk 2 above)
Comment (aaronenyeshi) [reply tests/Unit/Design/pass_by_ref.cpp:27]: Okay sounds good to me

**Review by scchan — APPROVED** [review]
LGTM

**Comment by AlexVlx** [comment]
@scchan @aaronenyeshi laddies, can we progress this one / merge it? It would very much ease with the staging of the following steps, and make sure that we don't run afoul of the upcoming cutoff. Thank you.

**Comment by scchan** [comment]
@AlexVlx   there are still 3 unexpected failures, could you address them please?

********************
Failing Tests (3):
    HCC :: Unit/Copy/copy.cpp
    HCC :: Unit/Design/transpose.cpp
    HCC :: Unit/Indexing/tile_index.cpp

**Comment by AlexVlx** [comment]
No they are expected, I think I already explained this over Slack? Could you please review that discussion?

**Pull request merged**